samples/Calling/src/app/views/CallScreen.tsx

```diff
@@ -37,6 +37,12 @@ export const CallScreen = (props: CallScreenProps): JSX.Element => {
       });
       adapter.on('error', (e) => {
         console.error(e);
+        // TODO: Handle other error types from adapter event emitter. Error type does not
+        // have properties 'operation' and 'error'
+        // Do not call error handler when start screen sharing fails
+        if (e['operation'] === 'Call.startScreenSharing' && e['error']['_code'] === 400) {
```

Comment:
code style wise this is a little scary to assume these variables like this.

Can we bring them into their own variables and check the correct values?

Unfortunately people do look at this code or worse just copy and paste and quickly change and we have to set an example of best practices. When possible.

Reply:
Two points here.

1. We can and should introduce an interface for errors raised via events as well as in the adapter state that has the `operation` and `error` fields. I'll do it today and send a PR.
2. You will still have to dig around inside the `error` though, because we get a opaque `Error` from the base SDK -- typescript does not have a way to strongly type raised exceptions. That said, you should (a) not have to use a private field `_code`, there should be a public field; (b) use a function to duck-type and then typecast so that the rest of the code is clean. Similar to https://github.com/Azure/communication-ui-library/blob/e337a0813532da34430795921a4bfc324453d881/packages/react-composites/src/composites/CallComposite/adapter/AzureCommunicationCallAdapter.ts#L524 (or atleast wrap the duck-typing logic in a helper function, e.g https://github.com/Azure/communication-ui-library/blob/e48aefca725807e58d1d2e4a6ff0121f5b09437e/packages/chat-component-bindings/src/errorBarSelector.ts#L78)

Reply:
Here's a PR to add strong error types to adapters: https://github.com/Azure/communication-ui-library/pull/761

This renames `operation` -> `target`. So, if we like it, lets land that PR before this one.

Reply:
#761 has landed. Please update this PR to use the newly added types.